src/projections/Tools/Overview/OverviewPanel.java

```diff
@@ -342,55 +348,72 @@ protected void setRanges(SortedSet<Integer> selectedPEs, long startTime, long en
-
-		selectedPEs.size();
-
 		int numIntervals = endInterval - startInterval;
-		
-		entryData = new int[selectedPEs.size()][numIntervals];
 		float[][] idleData = new float[selectedPEs.size()][numIntervals];
 		float[][] utilizationData = new float[selectedPEs.size()][numIntervals];
-		
-		int pIdx=0;		
-		
-		for(Integer pe : selectedPEs){
-			readyReaders.add( new ThreadedFileReader(pe, intervalSize, myRun, 
-					startInterval, endInterval, entryData[pIdx], utilizationData[pIdx], idleData[pIdx]) );
-			pIdx++;
-		}
-		
-		// Pass this list of threads to a class that manages/runs the threads nicely
-		TimedProgressThreadExecutor threadManager = new TimedProgressThreadExecutor("Loading Overview in Parallel", readyReaders, this, true);
-		threadManager.runAll();
-		
-		// For historical reasons, we use a utilization range of 0 to 100
 		utilizationDataNormalized = new int[utilizationData.length][utilizationData[0].length];
 		idleDataNormalized = new int[idleData.length][idleData[0].length];
-
-		for (int i=0; i<utilizationData.length; i++) {
-			for (int j=0; j<utilizationData[i].length; j++) {
-				utilizationDataNormalized[i][j] = (int) (100.0f * utilizationData[i][j]);
-				idleDataNormalized[i][j] = (int) (100.0f * idleData[i][j]);
+		if (MainWindow.runObject[myRun].hasLogData()) {	
+			this.saveImage = saveImage;
+			mode = OverviewWindow.MODE_EP;
+			// Create a list of worker threads
+			LinkedList<Runnable> readyReaders = new LinkedList<Runnable>();
+			selectedPEs.size();
+			entryData = new int[selectedPEs.size()][numIntervals];
+			int pIdx=0;
+			for(Integer pe : selectedPEs){
+				readyReaders.add( new ThreadedFileReader(pe, intervalSize, myRun, 
+						startInterval, endInterval, entryData[pIdx], utilizationData[pIdx], idleData[pIdx]) );
+				pIdx++;
 			}
-		}
-		
-		// dispose of unneeded utilizationData
-		utilizationData = null;
-	
+			// Pass this list of threads to a class that manages/runs the threads nicely
+			TimedProgressThreadExecutor threadManager = new TimedProgressThreadExecutor("Loading Overview in Parallel", readyReaders, this, true);
+			threadManager.runAll();
+					// For historical reasons, we use a utilization range of 0 to 100		
+			for (int i=0; i<utilizationData.length; i++) {
+				for (int j=0; j<utilizationData[i].length; j++) {
+					utilizationDataNormalized[i][j] = (int) (100.0f * utilizationData[i][j]);
+					idleDataNormalized[i][j] = (int) (100.0f * idleData[i][j]);
+				}
+			}
+			utilizationData = null;
+			
 		// We default to coloring by entry method
-		colorByEntry();
+		//colorby util for sumdetail
+			colorByEntry();
+	
+
+	
+		} else if (MainWindow.runObject[myRun].hasSumDetailData()) {
+			
+			SortedSet<Integer> availablePEs = MainWindow.runObject[myRun].getValidProcessorList(ProjMain.SUMDETAIL);
+			int intervalSize =(int) MainWindow.runObject[myRun].getSumDetailIntervalSize();
+			int startInterval2=(int)startTime/intervalSize;
+			int endInterval2=(int)endTime/intervalSize - 1;
```

Comment:
I think this should be `(int) Math.ceil(((double)endTime) / intervalSize) - 1;`, so that end times that don't align with intervals get rounded up to use the containing interval.

Reply:
Changed